src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs

```diff
@@ -0,0 +1,405 @@
+            return Ok(());
+        }
+        self.agent
+            .pause_container(self.container_id.clone().into())
+            .await
+            .context("agent pause container")?;
+        Ok(())
+    }
+
+    pub async fn resume(&self) -> Result<()> {
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Running {
+            warn!(self.logger, "container is running no need to resume");
+            return Ok(());
+        }
+        self.agent
+            .resume_container(self.container_id.clone().into())
+            .await
+            .context("agent pause container")?;
+        Ok(())
+    }
+
+    pub async fn resize_pty(
+        &self,
+        process: &ContainerProcess,
+        width: u32,
+        height: u32,
+    ) -> Result<()> {
+        let logger = logger_with_process(process);
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Running {
+            warn!(logger, "container is running no need to resume");
+            return Ok(());
+        }
+        self.agent
+            .tty_win_resize(agent::TtyWinResizeRequest {
+                process_id: process.clone().into(),
+                row: height,
+                column: width,
+            })
+            .await
+            .context("resize pty")?;
+        Ok(())
+    }
+
+    pub async fn stats(&self) -> Result<Option<agent::StatsContainerResponse>> {
+        let stats_resp = self
+            .agent
+            .stats_container(self.container_id.clone().into())
+            .await
+            .context("agent stats container")?;
+        Ok(Some(stats_resp))
+    }
+
+    // to do
+    pub async fn update(&self, resources: &LinuxResources) -> Result<()> {
+        self.resource_manager
+            .update_cgroups(&self.config.container_id, Some(resources))
+            .await?;
+
+        let req = agent::UpdateContainerRequest {
+            container_id: self.container_id.container_id.clone(),
+            resources: resources.clone(),
+            mounts: Vec::new(),
+        };
+        self.agent
+            .update_container(req)
+            .await
+            .context("agent update container")?;
+        Ok(())
+    }
+}
+
+fn amend_spec(spec: &mut oci::Spec) -> Result<()> {
+    // hook should be done on host
+    spec.hooks = None;
+
+    if let Some(linux) = spec.linux.as_mut() {
+        linux.seccomp = None;
+
+        if let Some(resource) = linux.resources.as_mut() {
+            resource.devices = Vec::new();
+            resource.pids = None;
+            resource.block_io = None;
+            resource.hugepage_limits = Vec::new();
+            resource.network = None;
+        }
+
+        let mut ns: Vec<oci::LinuxNamespace> = Vec::new();
+        for n in linux.namespaces.iter() {
+            match n.r#type.as_str() {
+                oci::PIDNAMESPACE => continue,
```

Comment:
```rust
oci::PIDNAMESPACE | oci::NETWORKNAMESPACE => continue,
```

Reply:
Thanks for your review, has been updated.